### ISSUE TITLE
Multiple tags are treated as one when working on kdbx from KeepassXC

### ISSUE THREAD

**Issue by tporeba** [libkeepass/pykeepass#338]
I found a difference in how multiple tags  `entry.tags` is parsed when reading a kdbx file saved by [KeePass Password Safe](https://keepass.info/) and [KeepassXC](https://keepassxc.org/). 

Here is example snippet:

```
import sys
from pykeepass import PyKeePass


def print_entries_tags(entries):
    for entry in entries:
        print(f"title={entry.title}")
        print(f"username={entry.username}")
        print(f"password={entry.password}")
        print(f"tags ({len(entry.tags or [])})")
        for tag in (entry.tags or []):
            print(f"{tag}")


def main(argv):
    xc = PyKeePass(filename="test_keepassxc.kdbx", password="test")
    print("Database from KeepassXC ==========")
    print_entries_tags(xc.entries)

    print("Database from Keepass ==========")
    kp = PyKeePass(filename="test_keepass.kdbx", password="test");
    print_entries_tags(kp.entries)


if __name__ == "__main__":
    main(sys.argv[1:])

```
To run it, unzip the attached [dbs.zip](https://github.com/libkeepass/pykeepass/files/11202513/dbs.zip) in the same folder as the script, you should have `test_keepass.kdbx` and `test_keepassxc.kdbx` files in the folder. Password for both databases is "test".

Output is 
```
Database from KeepassXC ==========
title=title
username=username
password=password
tags (1)
tag1,tag2
Database from Keepass ==========
title=title
username=username
password=password
tags (2)
tag1
tag2
```

Tags from KeepassXC database are treated as a single tag instead of 2 separate tags. 
Those programs can read each other's databases just fine and they both show that 2 separate tags exist on this entry. 

I used Python 3.11 and pykeepass 4.0.3

**Comment by tporeba** [comment]
It seems it is caused by different separators used by the two apps. 
I've inspected this fragment from `entry.py` using debugger:

```
@property
    def tags(self):
        val = self._get_subelement_text('Tags')
        return val.split(';') if val else val
```

KeepassXC uses comma `,` as separator and the `val` is "tag1,tag2", 
Keepass [uses semicolon](https://github.com/dlech/KeePass2.x/blob/a1555354f2e262cf185605426a79d216558f1243/KeePassLib/Utility/StrUtil.cs#L1529) `;` and `val` is "tag1;tag2". 

I'm not sure if I found the right places in their code, but it seems that when parsing, both are allowing extra separators: [KeepassXC ](https://github.com/keepassxreboot/keepassxc/pull/6487/files#diff-afad85487298147d34e500b13faa347e1afd9f4c5a9ebb578c9157256c66e1b1R193-R197), [Keepass](https://github.com/dlech/KeePass2.x/blob/a1555354f2e262cf185605426a79d216558f1243/KeePassLib/Utility/StrUtil.cs#L1441). 

Anyway, it would be nice for pykeepass to support both.

**Comment by Evidlo** [comment]
Should be fixed in 4.0.4.  Let me know if it works for you.
